sympy/functions/elementary/exponential.py

```diff
@@ -436,11 +436,21 @@ def _eval_nseries(self, x, n, logx):
         if arg0 in [-oo, oo]:
             return self
         t = Dummy("t")
-        exp_series = exp(t)._taylor(t, n)
+        nterms = n
+        try:
+            cf = Order(arg.as_leading_term(x), x).getn()
+        except NotImplementedError:
+            cf = 0
+        if cf and cf > 0:
+            nterms = (n/cf).ceiling()
+        exp_series = exp(t)._taylor(t, n, nterms)
         o = exp_series.getO()
         exp_series = exp_series.removeO()
         r = exp(arg0)*exp_series.subs(t, arg_series - arg0)
-        r += Order(o.expr.subs(t, (arg_series - arg0)), x)
+        if cf and cf > 0:
+            r += Order(o.expr.subs(t, (arg_series - arg0)), x)/x**((cf-1)*n)
```

Comment:
It looks like `o.expr` is `t**n`. It that is so, then it is not necessary to define `o` for this line.

Reply:
yes `o.expr` is `_t**n` but can you explain what do you mean by this (`not necessary to define o for this line`).

Reply:
I mean that if `o.expr` is equal to `t**n`, then there is no need to construct an order object `o` having this attribute. Instead of `o.expr.subs(t, (arg_series - arg0))` we can write `(arg_series - arg0)**n`. A new order object is created here anyway.

Reply:
okay..I have changed it.